Framework/Algorithms/src/ReflectometryReductionOneAuto.cpp

```diff
@@ -552,9 +589,13 @@ bool ReflectometryReductionOneAuto::processGroups() {
         AnalysisDataService::Instance().retrieveWS<Workspace>(firstTrans);
     firstTransG = boost::dynamic_pointer_cast<WorkspaceGroup>(firstTransWS);
 
-    if (!firstTransG)
+    if (!firstTransG) {
+      // we only have one transmission workspace, so we use it as it is.
       alg->setProperty("FirstTransmissionRun", firstTrans);
-    else if (group->size() != firstTransG->size())
+    } else if (group->size() != firstTransG->size() &&
+               !isPolarizationCorrectionOn)
+      // if they are not the same size then we cannot associate a transmission
+      // group workspace member with every input group workpspace member.
```

Comment:
This is correct code, but rather a matter of personal choice. Although you have a single statement here, the effective block size is 5 lines(with the comment). I find it easier to read when curly backets are used. But as I said this is a matter or personal choice and maybe there are good reasons against it.